### PR TITLE
Added Filter

### PR DESCRIPTION
**Pull request by Ankur9669**
Features Added:
1. Filter by category
2. Filter by search

Files To Check:
1. Chip.js
2. ChipContainer.js
3. category-context.js
4. VideosContainer.js

### CODE REVIEW

File: FitVid/src/index.js [hunk 1]
```diff
@@ -11,29 +11,32 @@ import { HistoryProvider } from "./Context/history-context";
...
+          <WatchLaterProvider>
+            <HistoryProvider>
+              <LikedProvider>
+                <UserProvider>
+                  <ToastProvider>
+                    <Router>
+                      <App />
+                    </Router>
+                  </ToastProvider>
+                </UserProvider>
```
Comment (manojsarna) [inline FitVid/src/index.js]: Suggestion : you can place Router at the top as pe rthe best practice

File: FitVid/src/Pages/VideoListing/VideosContainer/VideosContainer.js [hunk 2]
```diff
@@ -2,23 +2,45 @@ import React, { useEffect, useState } from "react";
...
+      filteredVideos = videos.filter(
+        (video) =>
+          video.categoryName.toLowerCase().includes(search.toLowerCase()) ||
+          video.title.toLowerCase().includes(search.toLowerCase()) ||
+          video.channelName.toLowerCase().includes(search.toLowerCase())
+      );
+    }
+    if (category !== "All") {
+      filteredVideos = filteredVideos.filter(
+        (video) => video.categoryName === category
```
Comment (manojsarna) [inline FitVid/src/Pages/VideoListing/VideosContainer/VideosContainer.js]: Suggestion : Keep this in a  separate utility function

**Review by manojsarna — APPROVED** [review]
Overall the code is good

Do Chek the Commentss

File: FitVid/src/index.js (hunk 1 above)
Comment (Ankur9669) [reply FitVid/src/index.js]: Done

File: FitVid/src/Pages/VideoListing/VideosContainer/VideosContainer.js (hunk 2 above)
Comment (Ankur9669) [reply FitVid/src/Pages/VideoListing/VideosContainer/VideosContainer.js]: Done

File: FitVid/src/Context/category-context.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { createContext, useContext, useState } from "react";
+// import { historyReducer } from "../Reducer/history-reducer";
```
Comment (nikhiljugale007) [inline FitVid/src/Context/category-context.js]: suggestion: remove commented code.

File: FitVid/src/index.js [hunk 4]
```diff
@@ -11,29 +11,32 @@ import { HistoryProvider } from "./Context/history-context";
...
+                      <App />
+                    </ToastProvider>
+                  </UserProvider>
+                </LikedProvider>
+              </HistoryProvider>
+            </WatchLaterProvider>
+          </PlayListModalProvider>
+        </PlayListsProvider>
+      </CategoryProvider>
+    </Router>
```
Comment (nikhiljugale007) [inline FitVid/src/index.js:39]: This looks so cool 🚀.

**Review by nikhiljugale007 — APPROVED** [review]
Did not find any issues.
Good to merge.
Suggestion: You can change title from react app to your app name.

File: FitVid/src/index.js (hunk 4 above)
Comment (Ankur9669) [reply FitVid/src/index.js:39]: Cool kaise?😂

Comment (nikhiljugale007) [reply FitVid/src/index.js:39]: Means it is very clear what each line is doing and also looks like a arrow.😂

File: FitVid/src/Components/ChipsContainer/Chip/Chip.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { useCategory } from "../../../Context/category-context";
+import "./chip.css";
+
+const Chip = ({ title, id }) => {
+  const { category, setCategory } = useCategory();
+  return (
+    <div
+      className={category === title ? "chip chip-active" : "chip"}
```
Comment (gauravsinhaweb) [inline FitVid/src/Components/ChipsContainer/Chip/Chip.js:9]: nice use of state in classname

File: FitVid/src/Context/category-context.js (hunk 3 above)
Comment (gauravsinhaweb) [inline FitVid/src/Context/category-context.js]: remove commented code for clean code!

**Review by gauravsinhaweb — APPROVED** [review]
1. context looks so clean and readable.
2. good use of states for className update.

some minor changes I suggested, Good to merge after changes.

File: FitVid/src/Context/category-context.js (hunk 3 above)
Comment (Ankur9669) [reply FitVid/src/Context/category-context.js]: Done

Comment (Ankur9669) [reply FitVid/src/Context/category-context.js]: Done

**Pull request merged**